com.unity.mobile.notifications/Editor/AndroidNotificationPostProcessor.cs

```diff
@@ -260,6 +262,34 @@ internal static void AppendAndroidMetadataField(string manifestPath, XmlDocument
 
             applicationNode.AppendChild(metaDataNode);
         }
+
+        private static void InjectProguard(string projectPath)
+        {
+            var proguardFile = $"{projectPath}/proguard-unity.txt";
+            if (!File.Exists(proguardFile))
+            {
+                UnityEngine.Debug.LogWarning($"Proguard file {proguardFile} not found, mobile notifications package may not function");
+                return;
+            }
+
+            var lines = File.ReadAllLines(proguardFile);
+            string[] newLines;
+            if (InjectProguard(lines, out newLines))
+                File.WriteAllLines(proguardFile, newLines);
+        }
+
+        internal static bool InjectProguard(string[] original, out string[] result)
+        {
+            foreach (var s in original)
+                if (s.Contains("com.unity.androidnotifications.UnityNotificationManager"))
```

Comment:
wouldn't it be more correct to search for presence of the string you're adding it, meaning
```
"-keep class com.unity.androidnotifications.UnityNotificationManager { public *; }"
```

Reply:
An existing project may already have this setup, so I'd rather search for class.

Reply:
Alright, hopefully nobody will add an entry like com.unity.androidnotifications.UnityNotificationManagerMyEx